src/librustdoc/clean/mod.rs

```diff
@@ -1211,8 +1211,14 @@ fn resolve(cx: &DocContext, path_str: &str, is_val: bool) -> Result<(Def, Option
                 } else {
                     match cx.tcx.type_of(did).sty {
                         ty::TyAdt(def, _) => {
-                            if let Some(item) = def.all_fields()
-                                                   .find(|item| item.ident.name == item_name) {
+                            if let Some(item) = if def.is_enum() {
+                                def.all_fields().find(|item| item.ident.name == item_name)
```

Comment:
This isn't a change that needed to be reverted, unless you wanted to iterate through `def.variants` here instead of `def.all_fields()`.

Reply:
Yes indeed!